### ISSUE TITLE
PushEstimator object has no attribute solution

### ISSUE THREAD

**Issue by a0s** [erp12/pyshgp#135]
`python examples/string_demo.py` doesn't work:

```
2020-05-29 11:10:42,186 INFO GENERATION: 1 | ERRORS: best=54, median=inf, diversity=0.04
2020-05-29 11:10:43,018 INFO GENERATION: 2 | ERRORS: best=54, median=59.0, diversity=0.085
2020-05-29 11:10:43,885 INFO GENERATION: 3 | ERRORS: best=40, median=54.0, diversity=0.085
2020-05-29 11:10:44,944 INFO GENERATION: 4 | ERRORS: best=40, median=56.0, diversity=0.105
2020-05-29 11:10:46,074 INFO GENERATION: 5 | ERRORS: best=35, median=59.0, diversity=0.18
2020-05-29 11:10:47,656 INFO GENERATION: 6 | ERRORS: best=28, median=54.0, diversity=0.295
2020-05-29 11:10:49,136 INFO GENERATION: 7 | ERRORS: best=28, median=54.0, diversity=0.215
2020-05-29 11:10:50,475 INFO GENERATION: 8 | ERRORS: best=19, median=54.0, diversity=0.205
2020-05-29 11:10:51,773 INFO GENERATION: 9 | ERRORS: best=19, median=54.0, diversity=0.17
2020-05-29 11:10:53,197 INFO GENERATION: 10 | ERRORS: best=19, median=54.0, diversity=0.245
2020-05-29 11:10:54,901 INFO GENERATION: 11 | ERRORS: best=19, median=54.0, diversity=0.26
2020-05-29 11:10:56,429 INFO GENERATION: 12 | ERRORS: best=19, median=54.0, diversity=0.275
2020-05-29 11:10:57,867 INFO GENERATION: 13 | ERRORS: best=19, median=54.0, diversity=0.19
2020-05-29 11:10:59,385 INFO Unsimplified solution found.
2020-05-29 11:10:59,385 INFO Simplifying genome of length 52.
2020-05-29 11:10:59,393 INFO Simplified to length 51.
2020-05-29 11:10:59,399 INFO Simplified to length 50.
2020-05-29 11:10:59,405 INFO Simplified to length 48.
2020-05-29 11:10:59,412 INFO Simplified to length 47.
2020-05-29 11:10:59,421 INFO Simplified to length 45.
2020-05-29 11:10:59,427 INFO Simplified to length 44.
2020-05-29 11:10:59,433 INFO Simplified to length 43.
2020-05-29 11:10:59,439 INFO Simplified to length 40.
2020-05-29 11:10:59,445 INFO Simplified to length 39.
2020-05-29 11:10:59,456 INFO Simplified to length 37.
2020-05-29 11:10:59,463 INFO Simplified to length 36.
2020-05-29 11:10:59,475 INFO Simplified to length 34.
2020-05-29 11:10:59,481 INFO Simplified to length 33.
2020-05-29 11:10:59,487 INFO Simplified to length 32.
2020-05-29 11:10:59,493 INFO Simplified to length 31.
2020-05-29 11:10:59,499 INFO Simplified to length 30.
2020-05-29 11:10:59,505 INFO Simplified to length 27.
2020-05-29 11:10:59,510 INFO Simplified to length 25.
2020-05-29 11:10:59,518 INFO Simplified to length 24.
2020-05-29 11:10:59,523 INFO Simplified to length 22.
2020-05-29 11:10:59,527 INFO Simplified to length 20.
2020-05-29 11:10:59,537 INFO Simplified to length 19.
2020-05-29 11:10:59,542 INFO Simplified to length 18.
2020-05-29 11:10:59,550 INFO Simplified to length 17.
2020-05-29 11:10:59,559 INFO Simplified to length 16.
2020-05-29 11:10:59,567 INFO Simplified to length 13.
2020-05-29 11:10:59,589 INFO Simplified to length 12.
2020-05-29 11:10:59,599 INFO Simplified to length 11.
2020-05-29 11:10:59,651 INFO Simplified to length 9.
2020-05-29 11:10:59,663 INFO Simplified to length 8.
2020-05-29 11:10:59,683 INFO Simplified to length 7.
2020-05-29 11:10:59,805 INFO Simplified to length 5.
2020-05-29 11:11:00,358 INFO Simplified genome length 5.
2020-05-29 11:11:00,358 INFO Simplified genome total error 0.
2020-05-29 11:11:00,358 INFO Simplified solution found.
Traceback (most recent call last):
  File "examples/string_demo.py", line 73, in <module>
    print(est.solution.get_program())
AttributeError: 'PushEstimator' object has no attribute 'solution'
```

python --version
Python 3.7.6

**Comment by erp12** [comment]
Thanks for reporting this. Did you install `pyshgp` from pip or from a clone of the repo?

**Comment by a0s** [comment]
@erp12 `pyshgp` was installed with pip

**Comment by erp12** [comment]
@a0s Thanks for the additional info. 

There was some breaking changes between version 0.1.5 and 0.1.6. At the time you installed from `pip` the latest published version of 0.1.5. but the `master` branch (I am guessing that is where you got the `string_demo.py` source) was already nearing 0.1.6.

If you switch branches in the GitHub UI to the branch/tag associated with the release you have installed the example files should work.

Hopefully we can settle the API soon and there will be no more breaking changes. Sorry about that. Also, I highly recommend upgrading to `pyshgp` v0.1.6 (released earlier today) because it is easier to use and some pretty nasty bugs were fixed.

**Comment by erp12** [comment]
I am assuming that my previous comment addressed this issue. If you still see any issues when running the examples on their corresponding `pyshgp` versions, feel free to reopen.
